pinot-common/src/main/antlr4/com/linkedin/pinot/pql/parsers/PQL2.g4

```diff
@@ -87,6 +88,9 @@ comparisonOperator: '<' | '>' | '<>' | '<=' | '>=' | '=';
 betweenClause:
   expression BETWEEN expression AND expression;
 
+regexClause:
+  expression (LIKE | REGEXP ) '(' literal (',' literal)* ')';
```

Comment:
It would be good to not require parentheses for this. For example, in pretty much every SQL database, the syntax `foo LIKE 'a%'` is valid.

This also copies the (non-standard) `REGEXP` keyword from MySQL. Unfortunately, that extension doesn't support adding regexp flags (eg. case insensitive matching, multiline match, etc.). I took a quick look at how other databases do it:

- MySQL: `foo REGEXP 'a.*'`, no support for regular expression flags
- PostgreSQL: `foo ~ 'a.*'` for case sensitive match, `foo ~* 'a.*'` for case insentitive match, 
- MSSQL: Not supported
- Oracle: `REGEXP_LIKE(foo, 'a.*', 'i')` with flag support (eg. `'i'` for case insensitive match)
- H2: Same as Oracle
- SQLite: Same as MySQL

Even though I normally like how PostgreSQL does things, in this case I think it's a bit weird and prefer the Oracle syntax (implement it as a function, with arguments, which also allows regexp flags).

Alternatively, if we use the MySQL (or PostgreSQL) syntax, we could encode the flags within the expression itself, just like Java does (eg. `foo(?im)` is a case-insensitive multiline match). Let's figure out which one we want, because once it's in the grammar we'll be stuck with it for a long time.